x-pack/legacy/plugins/apm/server/routes/settings/agent_configuration.ts

```diff
@@ -22,7 +22,7 @@ import { markAppliedByAgent } from '../../lib/settings/agent_configuration/mark_
 export const agentConfigurationRoute = createRoute(core => ({
   path: '/api/apm/settings/agent-configuration',
   handler: async req => {
-    const setup = await setupRequest(req);
+    const setup = await setupRequest(req, { clientAsInternalUser: true });
```

Comment:
Down the line do you think `setupRequest` should also expose a savedObjects client which takes `clientAsInternalUser` into account?

Reply:
Maybe! But I'm not yet 100% sold on the one-role-per-request approach yet 😬